### PR TITLE
feat: add FormGroup component

### PR DESCRIPTION
**Pull request by dierat**
### Summary:
Clubhouse ticket: https://app.shortcut.com/czi-edu/story/169595/button-group-form-group-move-to-eds
Location in `traject`: https://github.com/FB-PLP/traject/tree/master/app/assets/javascripts/v2/core/EDSCandidates/FormGroup

### Screenshots

FormGroup in storybook

### Test Plan:
- [x] Manually tested the component in storybook
- [x] Verified the snapshot tests output looks right

### CODE REVIEW

File: packages/components/src/FormGroup/index.ts
Comment (jeremiah-clothier) [inline packages/components/src/FormGroup/index.ts:1]: Should this be removed?

```suggestion

```

**Comment by jeremiah-clothier** [comment]
Is it intentional that this is being merged into `diedra/button-group/add` instead of the master/main branch?
image

File: packages/components/src/FormGroup/FormGroup.tsx
Comment (ahuth) [inline packages/components/src/FormGroup/FormGroup.tsx:71]: Does this result in HTML like with an extra P tag:

```html
<legend>
  <p class="stuff">Some text</p>
</legend>
```

?

If so, I wonder if it's worth thinking about reducing unnecessary elements. Maybe with something like

```jsx
<Text as={Component} className={clsx(className, styles.title)} size="sm" weight="bold">
  {children}
</Text>
```

Although that may not really be worth it. And would probably require updating the Text component to accept more things for `as`.

File: packages/components/src/FormGroup/FormGroup.tsx
Comment (ahuth) [inline packages/components/src/FormGroup/FormGroup.tsx:78]: I'm surprised to see the `Component` annotation here, since `ItemProps` defines `as` as `ElementType`.

Do we need that because TypeScript isn't happy with trying to do `<Component>` with intrinsic JSX elements or something?

Comment (jeremiah-clothier) [reply packages/components/src/FormGroup/FormGroup.tsx:78]: In this instance I believe `as: Component` is just renaming the variable and not providing a type annotation. Then you can use `<Component .../>` instead of `<as ... />`

Comment (ahuth) [reply packages/components/src/FormGroup/FormGroup.tsx:78]: Yep, you're right. It's not even possible to have a type annotation in the middle of object destructuring like that.

**Comment by dierat** [comment]
> Is it intentional that this is being merged into `diedra/button-group/add` instead of the master/main branch?

Yeah; I stacked them so it was easier to make an alpha version with both of them on it. I'll merge them into `main` separately.

File: packages/components/src/FormGroup/FormGroup.tsx
Comment (dierat) [reply packages/components/src/FormGroup/FormGroup.tsx:71]: https://app.shortcut.com/czi-edu/story/182120/allow-text-component-to-use-more-elements-for-as-prop

**Review by ahuth — APPROVED** [review]
LGTM

**Comment by dierat** [comment]
Any idea what's causing this?
screenshot of github actions post above stating that this PR is increasing the package size by 1.37% and needs changed

**Comment by ahuth** [comment]
No idea why a size increase of 1.37% would cause the size limit bot to block merging.

Couple options:
1. [Easiest] Dismiss the bot's review
2. Delete the size-limit-action. I don't think we really need it.
3. See if we can tune it's parameters

**Comment by dierat** [comment]
> No idea why a size increase of 1.37% would cause the size limit action to block merging. Maybe it's enforcing a total size limit of the package or something.
> 
> Couple options:
> 
> 1. [Easiest] Dismiss the action's review, and see if subsequent PRs have this same issue
> 2. Delete the size-limit-action. I don't think we really need it.
> 3. See if we can tune it's parameters

Cool, I'll just dismiss it for now. Thanks for verifying! 👍

**Pull request merged**
